### PR TITLE
[Bugfix, New Features] Ensure penalty sampler to be the first one in mixed samplers, accelerate TikToken, introduce pd disaggregation and separate acceleration on CPU backend

### PR DESCRIPTION
**Pull request by huangzhengxiang**
1. Ensure penalty sampler to be the first one in mixed samplers. Penalty shall apply to the original logits.
2. Accelerate `TikToken`, leveraging `Trie` data structure.

| prompt length | original O(n^3) | Trie O(n) |
|----------|----------|----------|
| 868 | 1.4s | 1ms |
| 24429 | 1h | 34ms |

(data on Snapdragon 8 Gen 3)

3. Introduce pd disaggregation and separate acceleration on CPU backend.
- Prefill and decode now can have separate configurations.
- CPU prefill becomes 15%～100% faster across 9 devices.

To name a few:

| device | SoC | original | new | speedup |
|----------|----------|----------|----------|----------|
| Huawei Mate 40 Pro | Kirin 9000 | 102 | 117 | 15% |
| Huawei Magic 6 | Snapdragon 8Gen3 | 148 | 285 | 92% |
| Xiaomi 15 Pro | Snapdragon 8 Elite | 258 | 368 | 42% |

(speed unit: token/s, tested on Qwen2.5-1.5B int4 quant, block=128)

### CODE REVIEW

File: express/module/Module.cpp [hunk 1]
```diff
@@ -268,6 +268,22 @@ class NetModule : public Module {
         NetModule* module(new NetModule(submodule, newInfo, nullptr, 0, 0.0f));
 #ifdef MNN_INTERNAL_ENABLED
         module->mLogInfo = mLogInfo;
+#endif
+        return this->cloneBaseTo(ctx, module);
+    }
+    virtual Module* clone(CloneContext* ctx, const ScheduleConfig* config) const override {
+        auto mModule = mChildren[0];
+        auto origin = mInfo->runTimeManager->getInside();
```
Comment (jxt1234) [inline express/module/Module.cpp:276]: 这里修改的原因是?

File: transformers/llm/engine/src/llm.cpp [hunk 2]
```diff
@@ -335,15 +356,55 @@ bool Llm::select_module(size_t index) {
...
+            }
+            setKVCacheInfo(0, getCurrentHistory());
+            reset();
+        }
+        mRuntimeManager->setHint(itp_type, prefer_candidate);
+        // load prefill module again to take effect! the following 2 lines can't be deleted!!
+        for (int v = 0; v < mPrefillModules.size(); ++v) {
+            mPrefillModules[v].reset(Module::clone(mPrefillModules[v].get()));
+        }
+        switchMode(Prefill);
```
Comment (jxt1234) [inline transformers/llm/engine/src/llm.cpp:358]: 这里加 return 即可，不用增加判断

File: transformers/llm/engine/src/tokenizer.cpp [hunk 3]
```diff
@@ -449,33 +449,13 @@ void Tiktoken::encode(const std::string& str, std::vector<int>& ids) {
...
-            // or the input text contains characters that the encoder doesn't know how to handle
-            std::cerr << "Error: No encoding found for the sequence starting at position " << i << " , symbol: " << str[i-2] << std::endl;
-            return;
+    auto it = str.begin();
+    while(it!=str.end()) {
+        auto last_it = it;
+        int token_id = encoder_.find(it, str.end());
+        if (token_id>=0) { ids.push_back(token_id); }
+        else {
+            std::cerr << "Error: No encoding found for the sequence" << std::string(last_it, it) << std::endl;
```
Comment (jxt1234) [inline transformers/llm/engine/src/tokenizer.cpp]: 换成 MNN_ERROR

File: express/module/Module.cpp (hunk 1 above)
Comment (huangzhengxiang) [reply express/module/Module.cpp:276]: Prefill module和decode module可以共享所有参数，却使用不同config，不同线程数，不同power mode

File: transformers/llm/engine/src/tokenizer.cpp (hunk 3 above)
Comment (huangzhengxiang) [reply transformers/llm/engine/src/tokenizer.cpp]: ok

File: transformers/llm/engine/src/llm.cpp (hunk 2 above)
Comment (huangzhengxiang) [reply transformers/llm/engine/src/llm.cpp:358]: ok

File: express/module/Module.cpp (hunk 1 above)
Comment (huangzhengxiang) [reply express/module/Module.cpp:276]: 属于是增加了一个接口，不影响原有功能。也在其他后端测过，都没啥问题

File: source/backend/cpu/CPURuntime.cpp [hunk 4]
```diff
@@ -38,6 +38,9 @@
 
 #include <algorithm>
 #include <string>
+#include <iostream>
+#include <fstream>
```
Comment (jxt1234) [inline source/backend/cpu/CPURuntime.cpp:42]: source/backend/cpu
source/core 下面文件不要使用 stream
